pkg/providers/common/common.go

```diff
@@ -30,8 +30,8 @@ func Configure() {
 }
 
 // PrepareTokensAndClient setup the client, usually for a UserInfo request
-func PrepareTokensAndClient(r *http.Request, ptokens *structs.PTokens, setProviderToken bool) (*http.Client, *oauth2.Token, error) {
-	providerToken, err := cfg.OAuthClient.Exchange(context.TODO(), r.URL.Query().Get("code"))
+func PrepareTokensAndClient(r *http.Request, ptokens *structs.PTokens, setProviderToken bool, opts ...oauth2.AuthCodeOption) (*http.Client, *oauth2.Token, error) {
+	providerToken, err := cfg.OAuthClient.Exchange(context.TODO(), r.URL.Query().Get("code"), opts...)
```

Comment:
instead of passing `opts` into all of the `UserInfo` calls and `PrepareTokensAndClient`, just test for the configuration item at the time of the `cfg.OAuthClient.Exchange` call and then build the URL with an `appendCodeChallenge` call here.

`Cfg.OAuthOpts` may need to be adjusted to be an array (or maybe map). I think `IndieAuth` and Google use Cfg.OAuthOpts